conn.go

```diff
@@ -13,6 +14,11 @@ import (
 	"time"
 )
 
+var (
+	errInvalidWriteTopic     = errors.New("writes should NOT set Topic on kafka.Message")
+	errInvalidWritePartition = errors.New("writes should NOT set Partition on kafka.Message")
```

Comment:
How about rephrasing to say “cannot” or “must not” so it’s more assertive? “should” is a conditional which communicates that it may be OK under some conditions.